cmd/cert-checker/main.go

```diff
@@ -119,12 +119,16 @@ func (c *certChecker) getCerts(unexpiredOnly bool) error {
 	}
 
 	initialID, err := c.dbMap.SelectInt(
-		"SELECT MIN(id) FROM certificates WHERE issued >= :issued AND expires >= :now",
+		"SELECT IFNULL(MIN(id), -1) FROM certificates WHERE issued >= :issued AND expires >= :now",
```

Comment:
Seems like it might be better to use [`SelectNullInt`](https://pkg.go.dev/gopkg.in/gorp.v2#DbMap.SelectNullInt), which returns a sentinel value if the result is null, rather than reinventing the sentinel value wheel ourselves.

Or `SelectOne`, which returns `ErrNoRows` if no rows are found, but has a worse interface for filling an integer with the result.

Reply:
Oh, and can we add a test for the case where the DB has no rows? `//cmd/cert-checker/main-test.go` already has tests which mock the DB's `SelectInt` call to return 99999; adding another mock which returns the sentinel value and another tests which uses that mock should be pretty fast.

Reply:
Sounds good. I'll get the test together and look at the SelectNullInt vs SelectOne options.

Reply:
I added a test making sure the sentinel value got set properly on an empty DB. We've already got a test that will fail if the initialID isn't set properly so it didn't seem necessary to add that.